terraform/create_state_bucket/dynamodb.tf

```
@@ -0,0 +1,11 @@
+resource "aws_dynamodb_table" "terraform-state-locks" {
+  name           = "ashley_terraform_state_locks${var.unique_suffix}"
```

I don't think the dynamo table name must be unique across all AWS instances like for s3 bucket name. You can maybe remove this suffix.

---

The [doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table#name) says : `this needs to be unique within a region.`
So I think it's going to be problematic to set a static name.

WDYT if I use the same name than the S3 bucket that have been generated?

```suggestion
  name           = aws_s3_bucket.state_bucket.bucket
```

---

On second thought, maybe I'm confusing `unique` (per customer) and `globally unique` (across all AWS customers) .
I'll get rid of this variable ^^